inspirehep/dojson/utils.py

```diff
@@ -104,3 +104,38 @@ def strip_empty_values(obj):
         return type(obj)(new_obj)
     else:
         return obj
+
+
+def remove_duplicates_from_list(l):
+    """Remove duplicates from a list preserving the order.
+
+    We might be tempted to use the list(set(l)) idiom,
+    but it doesn't preserve the order, which hinders
+    testability."""
+    seen = set()
```

Comment:
And @jalavik is right! I was thinking _performance_, but it's actually **worse**:

```
In [1]: import random

In [2]: def test_with_set():
   ...:     l = random.sample(range(5), 3)
   ...:     seen = set()
   ...:     result = []
   ...:     for el in l:
   ...:         if el not in seen:
   ...:             result.append(el)
   ...:             seen.add(el)
   ...:     return result
   ...:

In [3]: def test_without_set():
    l = random.sample(range(5), 3)
    result = []
    for el in l:
        if el not in result:
            result.append(el)
    return result
   ...:

In [4]: %timeit test_with_set()
The slowest run took 5.11 times longer than the fastest. This could mean that an intermediate result is being cached.
100000 loops, best of 3: 4.11 µs per loop

In [5]: %timeit test_without_set()
The slowest run took 4.30 times longer than the fastest. This could mean that an intermediate result is being cached.
100000 loops, best of 3: 3.72 µs per loop
```

(yay microbenchmarks!)